### ISSUE TITLE
MapTool 1.5.2 NullPointerException when try to shutdown server.

### ISSUE THREAD

**Issue by adventuremagic123** [RPTools/maptool#473]
**Describe the bug**
I tried to shutdown the MapTool 1.5.2 server and got the following exception:
```
java.lang.NullPointerException
	at net.rptools.maptool.util.UPnPUtil.closePort(UPnPUtil.java:225)
	at net.rptools.maptool.client.MapTool.disconnect(MapTool.java:1139)
	at net.rptools.maptool.client.MapTool.stopServer(MapTool.java:1016)
	at net.rptools.maptool.client.AppActions.disconnectFromServer(AppActions.java:2280)
	at net.rptools.maptool.client.AppActions$61.execute(AppActions.java:2271)
	at net.rptools.maptool.client.AppActions$ClientAction.actionPerformed(AppActions.java:3089)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
        ...
```

**To Reproduce**
1. Start the MapTool 1.5.2 server with shared FoW.
2. Try to shutdown the server.
3. Get the above exception.

**Expected behavior**
Expected no error message.  Under the same scenario with a copy of the same campaign, MapTool 1.4.1.8 has no trouble.

**Screenshots**

**MapTool Info**
- Version: 1.5.2
- Install: New

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 10

**Additional context**
==== MapTool Information ====
MapTool Version: 1.5.2
MapTool Home...: C:\Users\mark\.maptool-rptools
Max mem avail..: 3 GB
Max mem used...: 1 GB
Custom Property: -DMAPTOOL_DATADIR=.maptool-rptools

==== Java Information ====
Java Vendor.: Oracle Corporation
Java Home...: C:\Users\mark\AppData\Local\MapTool\runtime
Java Version: 10.0.1

==== OS Information ====
OS Name........: Windows 10
OS Version.....: 10.0
OS Architecture: amd64
PATH...........: C:\Users\mark\AppData\Local\MapTool\app;C:\Users\mark\AppData\Local\MapTool
Number of Procs: 4

==== User Information ====
User Name: mark
User Home: C:\Users\mark
User Dir.: C:\Users\mark\AppData\Local\MapTool\app

==== Network Interfaces ====
Display Name..: Software Loopback Interface 1
Interface Name: lo
Address...: 127.0.0.1
Address...: 0:0:0:0:0:0:0:1

Display Name..: NETGEAR A6210 WiFi USB3.0 Adapter
Interface Name: wlan0
Address...: 192.168.1.100
Address...: fe80:0:0:0:e9f4:8d5:bb09:d330%wlan0

Host Address...: 192.168.1.100
Default Gateway: 192.168.1.1

==== Locale Information ====
Country.: United States
Language: English
Locale..: English (United States)
Variant.: 

==== Encoding Information ====
Default Locale:   en_US
Default Charset:  UTF-8
file.encoding:    UTF-8
sun.jnu.encoding: Cp1252
Default Encoding: UTF8

==== Display Information ====
Number of Displays: 3
Display 1: 2560x1440(32)
Display 2: 1600x900(32)
Display 3: 1600x900(32)

==== Internet Gateway Devices ====
Device Name.: R6400 (Gateway)
Model Name..: R6400
Manufacturer: NETGEAR, Inc.
Model Number: R6400
Model Desc..: NETGEAR R6400 802.11ac Wireless Router
Firmware....: Linux/2.6.12, UPnP/1.0, NETGEAR-UPNP/1.0
External IP.: 72.48.55.8

**Comment by Phergus** [comment]
This is a UPnP error.  MT was unable to close the port on the router.   Might check that your router firmware is up to date.  They had a new one in December.

**Comment by adventuremagic123** [comment]
MapTool 1.4.1.8 doesn't have any problem in the same scenario.

**Comment by adventuremagic123** [comment]
I mean  that as an FYI.

**Comment by adventuremagic123** [comment]
I will update the firmware and retest.

It has not been updated for a long time because I had a problem with an update -- but I can't remember what it was.

**Comment by adventuremagic123** [comment]
I updated to the latest V1.0.1.46_1.0.32 firmware version and still get an exception:

java.lang.NullPointerException
	at net.rptools.maptool.util.UPnPUtil.closePort(UPnPUtil.java:225)
	at net.rptools.maptool.client.MapTool.disconnect(MapTool.java:1139)
	at net.rptools.maptool.client.MapTool.stopServer(MapTool.java:1016)
	at net.rptools.maptool.client.AppActions.disconnectFromServer(AppActions.java:2280)
	at net.rptools.maptool.client.AppActions$61.execute(AppActions.java:2271)
	at net.rptools.maptool.client.AppActions$ClientAction.actionPerformed(AppActions.java:3089)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:369)
	at de.muntjak.tinylookandfeel.TinyMenuItemUI.doClick(Unknown Source)
	at de.muntjak.tinylookandfeel.TinyMenuItemUI$MouseInputHandler.mouseReleased(Unknown Source)
	at java.desktop/java.awt.Component.processMouseEvent(Component.java:6589)
	at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3342)
	at java.desktop/java.awt.Component.processEvent(Component.java:6354)
	at java.desktop/java.awt.Container.processEvent(Container.java:2261)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4966)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2319)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4798)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4914)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4543)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4484)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2305)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2772)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4798)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:772)
	at java.desktop/java.awt.EventQueue.access$600(EventQueue.java:97)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:97)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:745)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:743)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:87)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:742)
	at net.rptools.maptool.client.swing.MapToolEventQueue.dispatchEvent(MapToolEventQueue.java:43)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)

==== MapTool Information ====
MapTool Version: 1.5.2
MapTool Home...: C:\Users\mark\.maptool-rptools
Max mem avail..: 3 GB
Max mem used...: 1 GB
Custom Property: -DMAPTOOL_DATADIR=.maptool-rptools

==== Java Information ====
Java Vendor.: Oracle Corporation
Java Home...: C:\Users\mark\AppData\Local\MapTool\runtime
Java Version: 10.0.1

==== OS Information ====
OS Name........: Windows 10
OS Version.....: 10.0
OS Architecture: amd64
PATH...........: C:\Users\mark\AppData\Local\MapTool\app;C:\Users\mark\AppData\Local\MapTool
Number of Procs: 4

==== User Information ====
User Name: mark
User Home: C:\Users\mark
User Dir.: C:\Users\mark\AppData\Local\MapTool\app

==== Network Interfaces ====
Display Name..: Software Loopback Interface 1
Interface Name: lo
Address...: 127.0.0.1
Address...: 0:0:0:0:0:0:0:1

Display Name..: NETGEAR A6210 WiFi USB3.0 Adapter
Interface Name: wlan0
Address...: 192.168.1.100
Address...: fe80:0:0:0:e9f4:8d5:bb09:d330%wlan0

Host Address...: 192.168.1.100
Default Gateway: 192.168.1.1

==== Locale Information ====
Country.: United States
Language: English
Locale..: English (United States)
Variant.: 

==== Encoding Information ====
Default Locale:   en_US
Default Charset:  UTF-8
file.encoding:    UTF-8
sun.jnu.encoding: Cp1252
Default Encoding: UTF8

==== Display Information ====
Number of Displays: 3
Display 1: 2560x1440(32)
Display 2: 1600x900(32)
Display 3: 1600x900(32)

==== Internet Gateway Devices ====
Device Name.: R6400 (Gateway)
Model Name..: R6400
Manufacturer: NETGEAR, Inc.
Model Number: R6400
Model Desc..: NETGEAR R6400 802.11ac Wireless Router
Firmware....: Linux/2.6.12, UPnP/1.0, NETGEAR-UPNP/1.0
External IP.: 72.48.55.8

![Router Firmware Version Info](https://user-images.githubusercontent.com/50708405/57934450-61e6a600-7885-11e9-9bd1-cd18d51ee6f3.JPG)

**Comment by Phergus** [comment]
> MapTool 1.4.1.8 doesn't have any problem in the same scenario.

MT 1.5.2 may be running slightly faster and the router isn't responding soon enough.  Also, 1.4.1.8 used Java 8 which may have other differences.  I don't think the UPnP code in MT has changed since 1.4.1.8 but I'm going to take a look as the error is occurring at a weird place.

**Comment by Phergus** [comment]
> I updated to the latest V1.0.1.46_1.0.32 firmware version and still get an exception:

Thanks for checking.

**Comment by Phergus** [comment]
Found issue in the code.  The port does get unmapped and so while annoying it is basically meaningless.

Will be fixed in 1.5.3.

**Comment by Phergus** [comment]
Fixed with PR #475
